packages/formstr-app/src/provider/ProfileProvider.tsx

```diff
@@ -35,50 +42,128 @@ export const ProfileProvider: FC<ProfileProviderProps> = ({ children }) => {
   const [pubkey, setPubkey] = useState<string | undefined>(undefined);
   const [usingNip07, setUsingNip07] = useState(false);
   const [userRelays, setUserRelays] = useState<string[]>([]);
+  const [isGlobalRelayModalOpen, setGlobalRelayModalOpen] = useState(false);
 
   const { poolRef } = useApplicationContext();
 
-  const fetchUserRelays = async (pubkey: string) => {
+  const fetchUserRelaysFromNostr = useCallback(async (pubkey: string) => {
     if (!poolRef) return;
     let filter: Filter = {
       kinds: [10002],
       authors: [pubkey],
     };
     let relayEvent = await poolRef.current.get(getDefaultRelays(), filter);
-    if (!relayEvent) return;
+    if (!relayEvent) {
+      setUserRelays(getDefaultRelays());
+      return;
+    };
     let relayUrls = relayEvent.tags
       .filter((t) => t[0] === "r")
       .map((r) => r[1]);
     setUserRelays(relayUrls);
-  };
+  }, [poolRef]);
 
   useEffect(() => {
     const profile = getItem<IProfile>(LOCAL_STORAGE_KEYS.PROFILE);
     if (profile) {
       setPubkey(profile.pubkey);
-      fetchUserRelays(profile.pubkey);
+      fetchUserRelaysFromNostr(profile.pubkey);
     } else {
-      console.log("Couldn't find npub");
+      setUserRelays(getDefaultRelays());
     }
-  }, [poolRef]);
+  }, [poolRef, fetchUserRelaysFromNostr]);
 
   const logout = () => {
     setItem(LOCAL_STORAGE_KEYS.PROFILE, null);
     setPubkey(undefined);
+    setUserRelays(getDefaultRelays());
   };
 
   const requestPubkey = async () => {
     setUsingNip07(true);
     let publicKey = await window.nostr.getPublicKey();
     setPubkey(publicKey);
     setItem(LOCAL_STORAGE_KEYS.PROFILE, { pubkey: publicKey });
+    await fetchUserRelaysFromNostr(publicKey);
     setUsingNip07(false);
     return pubkey;
   };
 
+  const toggleGlobalRelayModal = () => {
+    setGlobalRelayModalOpen(prev => !prev);
+  };
+
+  const updateUserRelays = async (newRelays: string[]) => {
+    setUserRelays(newRelays);
+    message.success("Default relays updated.");
+
+    if (!pubkey) {
+        message.warning("Login to save your relay list to Nostr.");
+        return;
+    };
+
+    try {
+      const tags = newRelays.map(url => ['r', url]);
+      const event: UnsignedEvent = {
+        kind: 10002,
+        created_at: Math.floor(Date.now() / 1000),
+        tags,
+        content: "",
+        pubkey,
+      };
+      
+      const signedEvent = await window.nostr.signEvent(event);
+      const pool = new SimplePool();
+      await Promise.allSettled(pool.publish(newRelays, signedEvent));
+      pool.close(newRelays);
+      message.success("Relay list published to Nostr!");
+    } catch (error) {
+      console.error("Failed to publish NIP-65 event", error);
```

Comment:
would be better to throw the error as well, since the return value of this function is not needed.

The calling function would never if there was an error in this function or not